BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -504,45 +507,86 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
         // Allow a certificate if its status was evaluated successfully and the result is that it should be trusted
         BOOL shouldTrustServer = (status == errSecSuccess && (trustResult == kSecTrustResultProceed || trustResult == kSecTrustResultUnspecified));
 
-        [self completeServerTrustAuthenticationChallenge:challenge shouldTrust:shouldTrustServer];
+        if (shouldTrustServer) {
+            SecTrustRef serverTrust = [[challenge protectionSpace] serverTrust];
+            NSURLCredential *serverTrustCredential = [NSURLCredential credentialForTrust:serverTrust];
+            if (completionHandler) {
+                completionHandler(NSURLSessionAuthChallengeUseCredential, serverTrustCredential);
+                //FIXME: Or should we use NSURLSessionAuthChallengePerformDefaultHandling
```

Comment:
This was just for me to remember during development. I think it might work either way (the default handling ignores the credential, because it doesn't need it, it just does the default action).

But what I ended up going with is the closest analog to what we did before. I will remove the fixme.